lib/commands/actions.js

```diff
@@ -102,6 +107,75 @@ commands.unlock = function () {
   return androidHelpers.unlock();
 };
 
+commands.openNotifications = function () {
+  return this.bootstrap.sendAction("openNotification");
+};
+
+commands.setLocation = function (latitude, longitude) {
+  return this.adb.sendTelnetCommand(`geo fix ${longitude} ${latitude}`);
+};
+
+commands.pullFile = async function (remotePath) {
+  let localFile = temp.path({prefix: 'appium', suffix: '.tmp'});
+  await this.adb.pull(remotePath, localFile);
+  let data = await fs.readFile(localFile);
+  let b64data = new Buffer(data).toString('base64');
+  if (await fs.exists(localFile)) {
+    await fs.unlink(localFile);
+  }
+  return b64data;
+};
+
+commands.pushFile = async function (base64Data, remotePath) {
+  let localFile = temp.path({prefix: 'appium', suffix: '.tmp'});
+  await mkdirp(path.dirname(localFile));
+  let content = new Buffer(base64Data, 'base64');
+  let fd = await fs.open(localFile, 'w');
+  await fs.write(fd, content, 0, content.length, 0);
+  await fs.close(fd);
+
+  // adb push creates folders and overwrites existing files.
+  await this.adb.push(localFile, remotePath);
+  if (await fs.exists(localFile)) {
+    await fs.unlink(localFile);
+  }
+};
+
+commands.pullFolder = async function (remotePath) {
+  let localFolder = temp.path({prefix: 'appium'});
+
+  let bufferOnSuccess = function (buffer) {
+    logger.debug("Converting in-memory zip file to base64 encoded string");
+    return buffer.toString('base64');
```

Comment:
This callback returns the buffer, but the `pullFolder` function returns nothing. You'll have to create a promise and return it.